### PR TITLE
Batch Processor: Log Support

### PR DESCRIPTION
**Pull request by keitwb**
This mimics the same logic as metrics for logs in the batchprocessor.

### CODE REVIEW

File: processor/batchprocessor/batch_processor.go [hunk 1]
```diff
@@ -182,6 +183,13 @@ func (bp *batchProcessor) ConsumeMetrics(_ context.Context, md pdata.Metrics) er
 	return nil
 }
 
+// ConsumeLogs implements LogsProcessor
+func (bp *batchProcessor) ConsumeLogs(_ context.Context, ld pdata.Logs) error {
+	// First thing is convert into a different internal format
```
Comment (tigrannajaryan) [inline processor/batchprocessor/batch_processor.go:188]: The comment does not seem to reflect what the line that comes after it does.

File: processor/batchprocessor/batch_processor.go [hunk 2]
```diff
@@ -274,3 +287,44 @@ func (bm *batchMetrics) add(item interface{}) {
...
+func (bm *batchLogs) export(ctx context.Context) error {
+	return bm.nextConsumer.ConsumeLogs(ctx, bm.logData)
+}
+
+func (bm *batchLogs) itemCount() uint32 {
+	return bm.logCount
+}
+
+func (bm *batchLogs) size() int {
+	return bm.logData.ResourceLogs().Len()
```
Comment (tigrannajaryan) [inline processor/batchprocessor/batch_processor.go:312]: What is size() supposed to return?

Comment (keitwb) [reply processor/batchprocessor/batch_processor.go]: Based on the metrics batcher, the size in bytes of the internal data representation of logs.  I added a helper method to pdata to derive it without the batch processor having to convert to internal.

File: processor/batchprocessor/batch_processor.go (hunk 1 above)
Comment (keitwb) [reply processor/batchprocessor/batch_processor.go]: Fixed.

**Comment by keitwb** [comment]
> Can we also enable batcher in the e2e log tests where relevant (if the similar trace tests use the batcher)?

I'll look into this.  The trace correctness test uses batch processor.

File: processor/batchprocessor/batch_processor.go (hunk 2 above)
Comment (tigrannajaryan) [reply processor/batchprocessor/batch_processor.go]: Do we have batching by size in bytes? I thought we only support batching by the number of items. Can you point me to the code which does byte-based batching? I don't see anything mentioned in the config of the batcher.

Comment (keitwb) [reply processor/batchprocessor/batch_processor.go]: It is just used for telemetry at the detailed level to record how many bytes the batches are.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Comment by tigrannajaryan** [comment]
@keitwb  Please resolve the conflict.

**Comment by keitwb** [comment]
> > Can we also enable batcher in the e2e log tests where relevant (if the similar trace tests use the batcher)?
> 
> I'll look into this. The trace correctness test uses batch processor.

Yeah it is probably best to add e2e tests for logging (beyond the existing perf test) with another PR/issue as it looks quite involved.

**Pull request merged**